src/Configuration/test/ConfigServerBase.Test/ConfigServerHostedServiceTest.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.Memory;
+using Steeltoe.Extensions.Configuration.Placeholder;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Steeltoe.Extensions.Configuration.ConfigServer.Test;
+
+public class ConfigServerHostedServiceTest
+{
+    private static readonly MemoryConfigurationSource _fastTests = new () { InitialData = TestHelpers._fastTestsConfiguration };
+
+    [Fact]
+    public void Constructor_ThrowsOnNull()
+    {
+        Assert.Throws<ArgumentNullException>(() => new ConfigServerHostedService(null, null));
+    }
+
+#if NET6_0_OR_GREATER
+    [Fact]
+    public async Task ServiceConstructsAndOperatesWithConfigurationManager()
+    {
+        var configurationManager = new ConfigurationManager();
+        configurationManager.AddInMemoryCollection(TestHelpers._fastTestsConfiguration);
+        configurationManager.AddConfigServer();
+        var service = new ConfigServerHostedService(configurationManager, null);
+
+        await service.StartAsync(default);
+        await service.StopAsync(default);
+        Assert.True(true, "Service constructed, started and stopped without exception");
```

Comment:
@bart-vmware this look better?
```suggestion
        var startAction = async () => await service.StartAsync(default);
        var stopAction = async () => await service.StopAsync(default);

        await startAction.Should().NotThrowAsync("ConfigServerHostedService should start");
        await stopAction.Should().NotThrowAsync("ConfigServerHostedService should be stoppable");
```

Reply:
updated, better now?

Reply:
This is basically point 7 at https://ddoomen.medium.com/dennis-17-laws-of-test-driven-development-74122e7582c:
> Only have one scenario per test. So doing something, asserting the outcome, then doing something else and then again asserting the outcome should be avoided. It makes the test harder to read, and will make it skip the second or third scenario if the first one fails.